packages/schematics/angular/migrations/update-13/drop-ie-polyfills_spec.ts

```diff
@@ -0,0 +1,538 @@
+ * IE11 requires the following for NgClass support on SVG elements
+ */
+import 'classlist.js';
+      `.trim(),
+    );
+
+    const migrated = await schematicRunner
+      .runSchematicAsync('drop-ie-polyfills', {}, tree)
+      .toPromise();
+
+    expect(migrated.readContent('src/polyfills.ts').trim()).toBe(
+      `
+/***************************************************************************************************
+ * BROWSER POLYFILLS
+ */
+      `.trim(),
+    );
+  });
+
+  it('keeps the BROWSER POLYFILLS comment preceeding an unused polyfill', async () => {
+    tree.overwrite(
+      'src/polyfills.ts',
+      `
+/***************************************************************************************************
+ * BROWSER POLYFILLS
+ */
+
+/**
+ * IE11 requires the following for NgClass support on SVG elements
+ */
+// import 'classlist.js';  // Run \`npm install --save classlist.js\`.
+      `.trim(),
+    );
+
+    const migrated = await schematicRunner
+      .runSchematicAsync('drop-ie-polyfills', {}, tree)
+      .toPromise();
+
+    expect(migrated.readContent('src/polyfills.ts').trim()).toBe(
+      `
+/***************************************************************************************************
+ * BROWSER POLYFILLS
+ */
+      `.trim(),
+    );
+  });
+
+  it('ignores a `package.json` that does not include polyfill dependencies', async () => {
+    const packageJson = `
+{
+  "name": "ng-new",
+  "version": "0.0.0",
+  "dependencies": {
+    "@angular/core": "^13.0.0"
+  },
+  "devDependencies": {
+    "@angular/cli": "^13.0.0"
+  }
+}
+    `.trim();
+
+    tree.overwrite('package.json', packageJson);
+
+    const migrated = await schematicRunner
+      .runSchematicAsync('drop-ie-polyfills', {}, tree)
+      .toPromise();
+
+    expect(migrated.readContent('package.json')).toBe(packageJson);
+
+    // No `npm install` should be scheduled, nothing to remove.
+    expect(schematicRunner.tasks).toEqual([]);
+  });
+
+  it('uninstalls `classlist.js` and `web-animations-js` packages', async () => {
+    tree.overwrite(
+      'package.json',
+      `
+{
+  "name": "ng-new",
+  "version": "0.0.0",
+  "dependencies": {
+    "@angular/core": "^13.0.0",
+    "classlist.js": "^1.0.0"
+  },
+  "devDependencies": {
+    "@angular/cli": "^13.0.0",
+    "web-animations-js": "^1.0.0"
+  }
+}
+    `.trim(),
+    );
+
+    const migrated = await schematicRunner
+      .runSchematicAsync('drop-ie-polyfills', {}, tree)
+      .toPromise();
+
+    // Assert `package.json` no longer contains removed dependencies.
+    expect(migrated.readContent('package.json')).toBe(
+      `
+{
+  "name": "ng-new",
+  "version": "0.0.0",
+  "dependencies": {
+    "@angular/core": "^13.0.0"
+  },
+  "devDependencies": {
+    "@angular/cli": "^13.0.0"
+  }
+}
+    `.trim(),
+    );
+
+    // Assert that `npm install` is scheduled.
+    const taskNames = schematicRunner.tasks.map((task) => task.name);
+    expect(taskNames).toEqual(['node-package']);
```

Comment:
I wasn't sure how to test that the `npm install` task got scheduled, so I just looked for its name here. I'm not super happy with this though, so I'm open to suggestions for better ways to verify this.

Reply:
That is  pretty much the best option currently.